pkg/core/hooks/conn/util.go

```diff
@@ -40,6 +47,85 @@ func convertUnixNanoToTime(unixNano uint64) time.Time {
 	return time.Unix(seconds, nanoRemainder)
 }
 
+func isFiltered(logger *zap.Logger, req *http.Request, opts models.IncomingOptions) bool {
+	filtered := false
+
+	for _, filter := range opts.Filters {
+		if filter.Host != "" {
+			regex, err := regexp.Compile(filter.Host)
```

Comment:
this is overlaps with the isPassThrough function, any way to make the operation similar?

Reply:
The reason I redefined this is because of different kind of matching being done. The isPassThrough function only does host, path and port, and cannot do URL methods and headers. I would probably have to import that part of the repository to call it here. Is that fine?

Reply:
Sure you can import the function here as to avoid code duplication.